### ISSUE TITLE
Refactor `__init__` methods throughout

### ISSUE THREAD

**Issue by bwpriest** [LLNL/MuyGPyS#116]
Right now we are initializing lower level classes (`KernelFn`s and `Hyperparameters`) inside of `MuyGPS`. However, I believe that we want to modify this so that the user creates the lower lever classes directly, and passes them to `MuyGPS`. This will serve to both make the API more readable and make the code internals more maintainable. 

The proposed change would move the current creation logic from 
```
k_kwargs = {
    "kern": "matern",
    "metric": "l2",
    "eps": {"val": 1e-5},
    "nu": {"val": "log_sample", "bounds": (0.1, 5.0)},
    "length_scale": {"val": 1.0},
}
muygps = MuyGPS(**k_kwargs)
```
to something like
```
muygps = MuyGPS(
    Matern(
        disortion=Isotropic("l2"), 
        nu=Hyperparameter("log_sample", (0.1, 5.0))
        length_scale=Hyperparameter(1.0),
    ),
    eps=HomoscedasticNoise(1e-5),
    sigma_sq=SigmaSq(1.0),
)
```

**Comment by bwpriest** [comment]
Another advantage this provides is that model choices are _explicit_. Instead of depending implicitly upon conventions (e.g. `eps` type determined by the shape of the passed value), the user explicitly declares the model choices at intialization time.

**Comment by bwpriest** [comment]
Addressed with PR #118
